code/modules/mob/living/carbon/human/life.dm

```diff
@@ -62,6 +62,8 @@
 
 	//No need to update all of these procs if the guy is dead.
 	fall() //VORESTATION EDIT. Prevents people from floating
+	if(unnaturally_resized)

```

Comment:
Adding this to life means it ticks pretty often, and life is already overclogged, There's definitely a better place to do this. IIRC without looking- you can make this trigger every time someone enters or leaves the dorms areas.

Reply:
This was discussed in discord. If() clause which checks if variable is true is one of cheapest things possible and won't affect perfomance at all. At the same time, it pretty much guarantees that nothing will be abnormally-sized outside of dorms. While exited/entered check not only have cases where you can escape area at abnormal size (most of them related to legacy shitty code, in which loc is being directly set), in discord it was discussed that this method would consume even more resources overall.